TUnit.Core/Hooks/InstanceHookMethod.cs

```diff
@@ -22,7 +22,7 @@ public record InstanceHookMethod : IExecutableHook<TestContext>
 
     public TAttribute? GetAttribute<TAttribute>() where TAttribute : Attribute => Attributes.OfType<TAttribute>().FirstOrDefault();
 
-    public TimeSpan? Timeout => GetAttribute<TimeoutAttribute>()?.Timeout;
+    public TimeSpan? Timeout => GetAttribute<TimeoutAttribute>()?.Timeout ?? TimeSpan.FromMinutes(5);
```

Comment:
Instead of getting it from the attribute directly, the attribute should instead have its `IHookRegisteredEventReceiver` method invoked, and that should be used to override this property